iroh-sync/src/store/fs.rs

```diff
@@ -513,6 +518,68 @@ impl crate::ranger::Store<SignedEntry> for StoreInstance {
         let iter2 = RangeIterator::empty(&self.store.db)?;
         Ok(iter.chain(iter2))
     }
+
+    // TODO: Not horrible
```

Comment:
this should be fixed before merging

Reply:
Do you have an idea how to solve this efficiently without moving the complete database structure to a tree based one?

Reply:
I meant avoiding the allocations for now, not the algorithm itself, that will need more structural changes from the work @rklaehn will do

Reply:
I removed the allocations.